scripts/just-preset.js

```diff
@@ -0,0 +1,86 @@
+// @ts-check
+
+const { task, series, parallel, condition, option, argv, logger, addResolvePath } = require('just-scripts');
+
+const path = require('path');
+const fs = require('fs');
+
+const { clean } = require('./tasks/clean');
+const { copy } = require('./tasks/copy');
+const { jest, jestWatch } = require('./tasks/jest');
+const { sass } = require('./tasks/sass');
+const { ts } = require('./tasks/ts');
+const { tslint } = require('./tasks/tslint');
+const { webpack, webpackDevServer } = require('./tasks/webpack');
+const { verifyApiExtractor, updateApiExtractor } = require('./tasks/api-extractor');
+const lintImports = require('./tasks/lint-imports');
+const prettier = require('./tasks/prettier');
+const bundleSizeCollect = require('./tasks/bundle-size-collect');
+const checkForModifiedFiles = require('./tasks/check-for-modified-files');
+const generateVersionFiles = require('./tasks/generate-version-files');
+const perfTest = require('./tasks/perf-test');
+
+module.exports = function preset() {
+  // this add s a resolve path for the build tooling deps like TS from the scripts folder
+  addResolvePath(__dirname);
+
+  option('production');
+
+  // Adds an alias for 'npm-install-mode' for backwards compatibility
+  option('min', { alias: 'npm-install-mode' });
+
+  option('webpackConfig', { alias: 'w' });
+
+  // Build only commonjs (not other TS variants) but still run other tasks
+  option('commonjs');
+
+  task('clean', clean);
+  task('copy', copy);
+  task('jest', jest);
+  task('jest-watch', jestWatch);
+  task('sass', sass);
+  task('ts:commonjs', ts.commonjs);
+  task('ts:esm', ts.esm);
+  task('ts:amd', ts.amd);
+  task('tslint', tslint);
+  task('ts:commonjs-only', ts.commonjsOnly);
+  task('webpack', webpack);
+  task('webpack-dev-server', webpackDevServer);
+  task('verify-api-extractor', verifyApiExtractor);
+  task('update-api-extractor', updateApiExtractor);
+  task('lint-imports', lintImports);
+  task('prettier', prettier);
+  task('bundle-size-collect', bundleSizeCollect);
+  task('check-for-modified-files', checkForModifiedFiles);
+  task('generate-version-files', generateVersionFiles);
+  task('perf-test', perfTest);
+  task('ts', () => {
+    return argv().commonjs
+      ? 'ts:commonjs-only'
+      : parallel('ts:commonjs', 'ts:esm', condition('ts:amd', () => argv().production && !argv().min));
+  });
+
+  task('validate', series('tslint', () => (fs.existsSync('jest.setup.js') ? 'jest' : null)));
+  task('code-style', series('prettier', 'tslint'));
+  task('update-api', series('clean', 'copy', 'sass', 'ts', 'update-api-extractor'));
+  task('dev', series('clean', 'copy', 'sass', 'webpack-dev-server'));
+
+  // Special case build for the serializer, which needs to absolutely run typescript and jest serially.
+  task('build-jest-serializer-merge-styles', series('ts', 'jest'));
+
+  task('build:node-lib', series('clean', 'copy', parallel(condition('validate', () => !argv().min), 'ts:commonjs-only')));
```

Comment:
Starting jest simultaneously with build seems a bit weird. Even if it would technically work since jest compiles the code on its own, it seems like that could result in much harder to read output when there's a compiler error.

Reply:
Is this only in node-lib that you're concerned with here? If so I can switch it easily. Our node-libs tend to be really small.

Reply:
okay I made those go in series

Reply:
I actually meant all the presets, I just happened to comment on node-lib.

Reply:
i think we had it in parallel before...

Reply:
Oh hmm, it is in parallel right now